model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,369 @@
+  {
+    const auto parentComp =
+        _ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+    const auto nameComp =
+        _ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+    if(_jointName.length() && _jointName != nameComp->Data())
+        continue;
+
+    if (parentComp)
+    {
+      std::string parentInfo;
+      auto parentNameComp =
+          _ecm.Component<ignition::gazebo::components::Name>(
+          parentComp->Data());
+
+      if (parentNameComp)
+      {
+        parentInfo += parentNameComp->Data() + " ";
+      }
+      parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+
+      std::cout << "  - Joint [" << _entity << "]" << std::endl
+                << "    - Name: " << nameComp->Data() << std::endl
+                << "    - Parent: " << parentInfo << std::endl;
+    }
+
+    const auto jointComp =
+        _ecm.Component<ignition::gazebo::components::JointType>(_entity);
+    const auto childLinkComp =
+        _ecm.Component<ignition::gazebo::components::ChildLinkName>(_entity);
+    const auto parentLinkComp =
+        _ecm.Component<ignition::gazebo::components::ParentLinkName>(_entity);
+
+    if (childLinkComp && parentLinkComp)
+    {
+      std::cout << "    - Joint type:  " << jointTypes.at(jointComp->Data())
+      << "\n"   << "    - Parent Link: [" << childLinkComp->Data() << "]\n"
+                << "    - Child Link:  [" << parentLinkComp->Data() << "]\n";
+    }
+  }
+}
+}
+
+//////////////////////////////////////////////////
+extern "C" IGNITION_GAZEBO_VISIBLE void cmdModelList()
+{
+  ignition::gazebo::EntityComponentManager ecm{};
+  if(!PopulateECM(ecm))
+  {
+    return;
+  }
+  const auto models = ecm.EntitiesByComponents(
+    ignition::gazebo::components::ParentEntity(1),
+    ignition::gazebo::components::Model());
+
+  std::cout << "Available models:" << std::endl;
+
+  for(const auto &m : models)
+  {
+    const auto nameComp =
+        ecm.Component<ignition::gazebo::components::Name>(m);
+    std::cout << "    - " << nameComp->Data() << std::endl;
+  }
+}
+
+//////////////////////////////////////////////////
+extern "C" IGNITION_GAZEBO_VISIBLE void cmdModelInfo(
+  const char *_model, int _pose, int _link, const char *_linkName,
+  int _joint, const char *_joint_name)
+{
+  std::string linkName{""};
+  if(_linkName)
+    linkName = _linkName;
+  std::string jointName{""};
+  if(_joint_name)
+    jointName = _joint_name;
+
+  bool printAll{false};
+  if(!_pose && !_link && !_joint)
+    printAll = true;
+
+  if(!_model)
+  {
+    std::cerr << std::endl << "Model name not found" << std::endl;
+    return;
+  }
+  const std::string model{_model};
+  // Get arguments
+
+  ignition::gazebo::EntityComponentManager ecm{};
+  if(!PopulateECM(ecm))
+    return;
+
+  // Get the desired model entity.
+  auto entity = ecm.EntityByComponents(
+    ignition::gazebo::components::Name(model),
+    ignition::gazebo::components::Model());
```

Comment:
If the model is not found, the user currently doesn't have any indication of what happened.

You could check `if (kNullEntity == entity)` here and print something like "No model named <> was found".

Reply:
Addressed